x-pack/test_serverless/api_integration/test_suites/observability/fleet/fleet.ts

```diff
@@ -11,10 +11,45 @@ import { FtrProviderContext } from '../../../ftr_provider_context';
 export default function ({ getService }: FtrProviderContext) {
   const svlCommonApi = getService('svlCommonApi');
   const supertest = getService('supertest');
+  const retry = getService('retry');
 
-  // FLAKY: https://github.com/elastic/kibana/issues/176858
-  describe.skip('fleet', function () {
+  const defaultFleetServerHostId = 'default-fleet-server';
+  const defaultFleetServerHostUrl = 'https://localhost:8220';
+  const defaultElasticsearchOutputId = 'es-default-output';
+  const defaultElasticsearchOutputHostUrl = 'https://localhost:9200';
+
+  async function expectDefaultFleetServer() {
```

Comment:
is it possible to move `expectDefaultFleetServer` and `expectDefaultElasticsearchOutput` to a helper to avoid duplication?

Reply:
It's not super easy to read IMHO because of the different details (e.g. `body.item.host_urls` vs. `body.item.hosts`) 😅 maybe something like:

```js
async function expectDefault({ name, endpoint, defaultId, hostField, defaultHostUrl }) {
    await retry.waitForWithTimeout(`get default ${name}`, 30_000, async () => {
      const { body, status } = await supertest.get(`/api/fleet/${endpoint}/${defaultId}`);
      if (status === 200 && body.item[hostField].includes(defaultHostUrl)) {
        return true;
      } else {
        throw new Error(`Expected default ${name} id ${defaultId} to exist`);
      }
    });
  }
```

and

```js
const fleetServer = {
    name: 'Fleet Server',
    endpoint: 'fleet_server_hosts',
    defaultId: 'default-fleet-server',
    hostField: 'host_urls',
    defaultHostUrl: 'https://localhost:8220',
  };
```

Any thoughts or recommendation?

Reply:
Sorry, I didn't mean joining the 2 functions, but the duplication in `observability/fleet` and `security/fleet`.
The suggested refactor could make sense too, though not really necessary, I think.

Reply:
My bad, that makes a lot more sense 🙈 I wasn't sure as these two tests are identical anyway, but I tried to extract this and part of the config as they are linked. Let me know what you think.

Reply:
Looks great 👍

Reply:
Thanks for suggesting 👍 I've re-run the flaky test runner for good measure, all green: https://buildkite.com/elastic/kibana-flaky-test-suite-runner/builds/5237